giraffe/README.md

```diff
@@ -0,0 +1,341 @@
+
+- **legendCrosshairColor**: _string | Object. Optional._ The _CSS color value_ or styling of the vertical crosshair line through the Plot at where the mouse is hovering, defined as a [_CanvasRenderingContext2D strokeStyle_](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/strokeStyle).
+
+- **legendColumns**: _array[string, ...]. Optional._ When included, this array will determine which column key names that should be included in the legend (tooltip). If this option is included as an empty array, the legend will be empty.
+
+## Utility Functions
+
+Giraffe comes with utility functions.
+
+- **fromFlux**: _function(string)._ Takes a Flux CSV, converts it, and returns a `Table` used in the [table property](#data-properties) of the config.
+
+- **fromRows**: _function([Object, ...], Object)._ The first argument is an array of objects, each representing a row of data. The optional second argument describes the schema for the data. Returns a `Table` used in the [table property](#data-properties) of the config.
+
+- **newTable**: _function(number)._ The argument is a length for a newly created `Table` with no initial data that allows only columns equal to that length to be added. Returns the created `Table`.
+
+## LayerConfig
+
+- **LineLayerConfig**: _Object_. Maximum one per `<Plot>`. Properties are:
+
+  - **type**: _"line". **Required**_. Specifies that this LayerConfig and `<Plot>` is a line graph.
+
+  - **x**: _string. **Required**_. The column key name of the column that should be visualized on the x-axis.
+
+  - **y**: _string. **Required**_. The column key name of the column that should be visualized on the y-axis.
+
+  - **fill**: _array[string, ...]. Optional._ An array of column key names of column filters that should be visualized. If this option is not included, the data in the graph will be interpreted as belonging to a single column.
+
+  - **position**: _"overlaid" | "stacked". Optional._ Indicates whether the line graph's lines have no bearing on other lines (overlaid), or the lines are cumulatives of every line below it, ie [stacked](https://help.infragistics.com/Help/Doc/Silverlight/2011.1/CLR4.0/html/xamWebChart_Stacked_Line_Chart.html).
+
+  - **hoverDimension**: _"x" | "y" | "xy". Optional. Defaults to "x" when not included._ Indicates whether the legend (tooltip) should display all data points along the "x" axis or "y" axis during mouse hover. When "xy" is chosen, the legend will display for a single data point nearest the mouse rather than for all data points along an entire axis.
+
+  - **maxTooltipRows**: _number. Optional._ The maxium number of data rows to display in the legend (tooltip). Subject to screen size limitations. Scrolling not implemented.
+
+  - **interpolation**: _string. Optional._ The style of the path between two data points on the same line. For example, "linear" is a straight path between two data points.
```

Comment:
Is there a list of `interpolation` values somewhere? It might be useful to include a reference to that here

Reply:
Good call, we should definitely be thorough and include references. But the downside is that there is only 1 good reference which is d3. We are admitting we rely on d3 right in the docs if we link to d3.

Oh well, I guess we have to do that for now until we stop relying on d3.